File: front/src/app/Models/Bacteriology.ts

import { BacteriologyType, Germ, BacteriologyInterpretation } from './BacteriologyEnums';

export class Bacteriology {
  id!: number;
  date!: Date;
  type!: typeof BacteriologyType[keyof typeof BacteriologyType];
  germs!: number[];
  interpretation!: typeof BacteriologyInterpretation[keyof typeof BacteriologyInterpretation];
  comment!: string;
  consultationId!: number;
}

File: Services/MedicalReport/src/main/java/org/example/medicalreport/Models/AnalyseDTOs/Bacteriology.java
package org.example.medicalreport.Models.AnalyseDTOs;

import lombok.Data;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Data
public class Bacteriology {
    long id;
    LocalDate date;
    int type;
    List<Integer> germs = new ArrayList<>();
    int interpretation;
    String comment;
    long consultationId;
}
